File: app/src/main/java/rod/bailey/trafficatnsw/hazard/details/TextFieldListItemView.java
package rod.bailey.trafficatnsw.hazard.details;

import android.content.Context;
import android.widget.LinearLayout;
import android.widget.TextView;

import static android.graphics.Color.*;
import static android.graphics.Typeface.*;
import static android.text.TextUtils.TruncateAt.END;
import static android.view.Gravity.*;
import static android.view.ViewGroup.LayoutParams.*;
import static rod.bailey.trafficatnsw.util.DisplayUtils.*;

public class TextFieldListItemView extends LinearLayout {

	private static final int TEXT_SIZE_SP = 12;

	@SuppressWarnings("unused")
	private static final String TAG = TextFieldListItemView.class
			.getSimpleName();

	private final Context ctx;

	private TextView fieldNameTextView;

	private TextView fieldValueTextView;

	public TextFieldListItemView(Context ctx, String fieldName,
			String fieldValue) {
		super(ctx);

		this.ctx = ctx;

		fieldNameTextView = new TextView(ctx);
		fieldValueTextView = new TextView(ctx);

		// Field name
		fieldNameTextView.setText(fieldName);
		fieldNameTextView.setTextColor(BLACK);
		fieldNameTextView.setTypeface(DEFAULT);
		// Note that setTextSize() takes an argument in SP, not PX
		fieldNameTextView.setTextSize(TEXT_SIZE_SP);
		fieldNameTextView.setSingleLine();
		fieldNameTextView.setEllipsize(END);
		fieldNameTextView.setBackgroundColor(TRANSPARENT);

		LayoutParams nameLLP = new LayoutParams(
				WRAP_CONTENT, WRAP_CONTENT);
		fieldNameTextView.setLayoutParams(nameLLP);

		// Field value
		fieldValueTextView.setText(fieldValue);
		fieldValueTextView.setTextColor(DKGRAY);
		fieldValueTextView.setTypeface(DEFAULT);
		// Note that setTextSize() takes an argument in SP, not PX
		fieldValueTextView.setTextSize(TEXT_SIZE_SP);
		fieldValueTextView.setSingleLine();
		fieldValueTextView.setEllipsize(END);
		fieldValueTextView.setBackgroundColor(TRANSPARENT);

		// Field name aligned left, field value aligned right
		LayoutParams valueLLP = new LayoutParams(
				WRAP_CONTENT, WRAP_CONTENT);
		valueLLP.weight = 1;
		fieldValueTextView.setLayoutParams(valueLLP);
		fieldValueTextView.setGravity(RIGHT);

		setBackgroundColor(WHITE);
		int p = dp2Px(ctx, 5);
		setPadding(p,p,p,p);
		setOrientation(HORIZONTAL);
		
		addView(fieldNameTextView);
		addView(fieldValueTextView);
	}

	@Override
	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
		super.onMeasure(widthMeasureSpec, heightMeasureSpec);
		setMeasuredDimension(getDisplaySizePx(ctx).x, dp2Px(ctx,30));
	}
}

File: app/src/main/java/rod/bailey/trafficatnsw/hazard/details/view/TextFieldListItemView.kt
package rod.bailey.trafficatnsw.hazard.details.view

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.App
import org.androidannotations.annotations.EViewGroup
import org.androidannotations.annotations.ViewById
import rod.bailey.trafficatnsw.R

@EViewGroup(R.layout.list_item_hazard_detail_text_field)
open class TextFieldListItemView(ctx: Context,
							val fieldName: String,
							val fieldValue: String) : LinearLayout(ctx) {
	@ViewById(R.id.tv_hazard_detail_field_name)
	@JvmField
	var fieldNameTextView: AppCompatTextView? = null

	@ViewById(R.id.tv_hazard_detail_field_value)
	@JvmField
	var fieldValueTextView: AppCompatTextView? = null

	@AfterViews
	fun afterViews() {
		fieldNameTextView?.text = fieldName
		fieldValueTextView?.text = fieldValue
	}
}
